hathor/p2p/manager.py

```diff
@@ -592,7 +604,14 @@ def connect_to_if_not_connected(self, peer: UnverifiedPeer | PublicPeer, now: in
 
         assert peer.id is not None
         if peer.info.can_retry(now):
-            self.connect_to(self.rng.choice(peer.info.entrypoints), peer)
+            if self.enable_ipv6 and not self.disable_ipv4:
+                self.connect_to(self.rng.choice(peer.info.entrypoints), peer)
+            elif self.enable_ipv6 and self.disable_ipv4:
+                self.connect_to(self.rng.choice(peer.info.ipv6_entrypoints), peer)
+            elif not self.enable_ipv6 and not self.disable_ipv4:
+                self.connect_to(self.rng.choice(peer.info.ipv4_entrypoints), peer)
+            else:
+                raise ValueError('IPv4 is disabled and IPv6 is not enabled')
```

Comment:
You can use `assert_never()` here.

Reply:
This was introduced in Python 3.11, right? 

We are still supporting 3.10, AFAIK, so I guess we should avoid this?

Reply:
While it was added to the built-in `typing` in Python 3.11, it is available in `typing_extensions` for Python 3.10 too (as a backport), we've been using it already.

Reply:
Got it. Make sense. You can keep raising the exception.